dbdimp.c

```diff
@@ -687,14 +687,17 @@ static char *parse_params(
     /* LIMIT should be the last part of the query, in most cases */
     if (! limit_flag)
     {
-      /*
-        it would be good to be able to handle any number of cases and orders
-      */
-      if ((*statement_ptr == 'l' || *statement_ptr == 'L') &&
-          (!strncmp(statement_ptr+1, "imit ?", 6) ||
-           !strncmp(statement_ptr+1, "IMIT ?", 6)))
+      if (statement_ptr+4 < statement_ptr_end)
       {
-        limit_flag = TRUE;
+        char *s = statement_ptr;
+        if ((s[0] == 'l' || s[0] == 'L') &&
```

Comment:
`man strncasecmp 3` ?

Reply:
I thought about it, but:
1) It depends on current locale (LC_CTYPE)
2) Is not in C standard (only in POSIX; so not available in all Perl systems)

And explicit compare fix both problems.